document.go

```diff
@@ -7,110 +7,67 @@ import (
 	"errors"
 	"fmt"
 	"hash/crc32"
-	"hash/fnv"
-	"reflect"
 	"sort"
 
 	"github.com/golang/snappy"
 )
 
-/*
-Document represents the downloaded REST API document
-Internally its contents would be
-bool, for booleans
-float64, for all numbers
-string, for strings
-[]interface{} for slices
-Document for nested objects
-nil for nils
-*/
+// Document represents a single G API resource. Internally its contents
+// would be represented by Go primitives.
+//
+// bool          for booleans
+// float64       for numbers
+// string        for strings
+// []interface{} for arrays
+// Document      for nested objects
+// nil           for nulls
 type Document map[string]interface{}
 
 // New returns new Document
 func New() Document {
 	return make(Document)
 }
 
+// Keys is a convenience method to return the keys in this Document
+func (d Document) Keys() []string {
+	keys := make([]string, 0, len(d))
+	for k := range d {
+		keys = append(keys, k)
+	}
+	return keys
+}
+
+// KeysSorted is a convenience method to return the keys in this Document, in sorted
+// order
+func (d Document) KeysSorted() []string {
+	keys := d.Keys()
+	sort.Strings(keys)
+	return keys
+}
+
+// String satisifies the Stringer interface
 func (d Document) String() string {
 	data, err := json.MarshalIndent(d, "", "  ")
 	if err != nil {
-		return err.Error()
+		panic(err)
```

Comment:
```suggestion
		// Using `%#v` flag to avoid recursive calls to String
		return fmt.Sprintf("%#v", d)
```

Better not to panic.